src/sphinx_book_theme/header_buttons/launch.py

```diff
@@ -118,9 +118,10 @@ def add_launch_buttons(
         )
 
     if jupyterhub_url:
+        repo_esc = repo_url.replace(":", "%3A")
```

Comment:
A more general replacement is

```python
from urllib.parse import quote
repo_esc = quote(repo_url)
```

This will happen automatically if you use url-building functions to build the query, rather than by hand:

```python
from urllib.parse import urlencode
url_params = urlencode(dict(repo=repo_url, urlpath=f"{ui_pre}/{repo}/{path_rel_repo}", branch=branch), safe="/")
url = f"{jupyterhub_url}/hub/user-redirect/git-pull?{url_params}"

Reply:
Aha - I was missing the `safe="/"` kwarg - that makes all the difference to readability.

Reply:
it does! I don't know why `safe="/"` is the default for `urllib.parse.quote`, but not `urllib.parse.urlencode`, but it should be fine for our purposes here.

Reply:
@matthew-brett wanna give this a shot so that our implementation is a bit less special-casey ?

Reply:
Have done ...